pandas/plotting/_core.py

```diff
@@ -882,8 +882,13 @@ def _make_plot(self):
             kws = dict(ax=ax)
             if self.mpl_ge_1_3_1():
                 kws['label'] = c if c_is_column else ''
-            self.fig.colorbar(img, **kws)
-
+            cbar = self.fig.colorbar(img, **kws)
```

Comment:
Maybe refactor this and https://github.com/pandas-dev/pandas/pull/20446/files#diff-2b118bda866f3d626ceb6b529c62cd1aR932 to a method. They're identical, right? I guess aside from `**kws` vs. `ax=ax`?